### ISSUE TITLE
Fails to create a sandbox with NDB <= 7.4 series binary.

### ISSUE THREAD

**Issue by Mikiya** [datacharmer/dbdeployer#111]
**Describe the bug**
It is not possible to create a sandbox using NDB version mysqld, which major version is less than or equals to 7.4. The command in init_db is empty.

**To Reproduce**
$ dbdeployer deploy single 7.4.28 --skip-start --sandbox-binary=/home/mikiya/opt/ndb --binary-version=5.6.48 --sandbox-home=/home/mikiya/sandboxes/cluster_7_4_28 --sandbox-directory=sql1 --keep-server-uuid --port=10229 --my-cnf-options='server_id=1' --my-cnf-options='!include /home/mikiya/sandboxes/cluster_7_4_28/my_common.cnf' --bind-address=0.0.0.0 --skip-report-host --my-cnf-options='report_host=sql1' --flavor=ndb
cmd:    /home/mikiya/sandboxes/cluster_7_4_28/sql1/init_db
err:    exit status 127
stdout: Error installing database in /home/mikiya/sandboxes/cluster_7_4_28/sql1

stderr: /home/mikiya/sandboxes/cluster_7_4_28/sql1/init_db: line 36: --user=mikiya: command not found

init_db output: Error installing database in /home/mikiya/sandboxes/cluster_7_4_28/sql1

error creating sandbox: 'init_db failure: exit status 127'
$ tail -20 sql1/init_db 
    echo "This script should run only once."
    exit 0
fi

 \
     \
    --user=mikiya \
    --basedir=$BASEDIR \
    --datadir=$DATADIR \
    --tmpdir=$SBDIR/tmp 
exit_code=$?
if [ "$exit_code" == "0" ]
then
    echo "Database installed in $SBDIR"
else
    echo "Error installing database in $SBDIR"
    exit $exit_code
fi

**Expected behavior**
The command in the init_db script should not be empty.

**NO Screenshots**
If applicable, copy and paste the command and the output. Do not use screen shots, unless they add value to the report.  Text that can be edited is preferred to graphics.

**Environment:**
 - OS: Gentoo Linux
 - dbdeployer version 1.52.0
 - tarball full name (e.g. mysql-5.6.40-linux-glibc2.12-x86_64.tar.gz) mysql-cluster-gpl-7.4.28-linux-glibc2.12-x86_64.tar.gz
 - tarball origin (from where it was downloaded)  https://dev.mysql.com/downloads/cluster/

**Hardware: (if applicable)**
 - Free storage
 - Total RAM

**Additional context**

I suspect that the problem is caused because "mysqld --version" prints two versions; 5.6.48 and 7.4.28 for the example above.

**Comment by datacharmer** [comment]
Hi,
Thanks for reporting this problem.

There are a few problems with this issue. The most important one is that dbdeployer assumes that the underlying MySQL version is at least 5.7. In fact, it has been tested with NDB 7.6 and 8.0 only. I know for sure that it doesn't work with 7.4. I haven't tried with 7.5, and I will add a check to make sure that only the supported versions will be recognized.

Sorry about this problem. I didn't know that versions 7.4 and 7.5 are still maintained.

The second problem is that there are confusing options in your command. I mention it because, if you try with NDB 7.6 or 8.0, you will have similar problems.

`-----------------------------------------------------------`

Here is a roundup of what problems I see in your command:

```
    --sandbox-binary=/home/mikiya/opt/ndb
```

What does this directory contain? It should have at least a directory named
'7.4.28', containing the expanded files from mysql-cluster-gpl-7.4.28-linux-glibc2.12-x86_64.tar.gz

```
    --binary-version=5.6.48
```

If the above assumption is correct, this option is not needed, as it says
"the directory /gome/mikiya/opt/ndb/7.4.28" contains version 5.6.48


```
    --my-cnf-options='server_id=1'
```
there is an option `--server-id` that does this


```
    --my-cnf-options='!include /home/mikiya/sandboxes/cluster_7_4_28/my_common.cnf'
```

Depending on what is in this include file, anything can go wrong. 
dbdeplpoyer has an option `--my-cnf=file_name` to use an external my.cnf more
safely


```
    --flavor=ndb
```
Not needed for deployment

`-----------------------------------------------------------`

My recommendation is to use dbdeployer standard directories to avoid confusion. 
SANDBOX_BINARY=$HOME/opt/mysql (contains expanded tarballs)
SANDBOX_HOME=$HOME/sandboxes (contains deployed sandboxes)

**Comment by Mikiya** [comment]
Hi Giuseppe,

Thank you for your reply.

> There are a few problems with this issue. The most important one is that dbdeployer assumes that the underlying MySQL version is at least 5.7. In fact, it has been tested with NDB 7.6 and 8.0 only. I know for sure that it doesn't work with 7.4. I haven't tried with 7.5, and I will add a check to make sure that only the supported versions will be recognized.

I understand.

> Sorry about this problem. I didn't know that versions 7.4 and 7.5 are still maintained.

Yes. Even though 7.4 is under "extended support" phase, it's maintained still.

https://www.oracle.com/us/assets/lifetime-support-technology-069183.pdf

> The second problem is that there are confusing options in your command. I mention it because, if you try with NDB 7.6 or 8.0, you will have similar problems.
> 
> -----------------------------------------------------------
> 
> Here is a roundup of what problems I see in your command:
> 
> --sandbox-binary=/home/mikiya/opt/ndb
> 
> What does this directory contain? It should have at least a directory named
> '7.4.28', containing the expanded files from mysql-cluster-gpl-7.4.28-linux-glibc2.12-x86_64.tar.gz

Yes. There is the "7.4.28" directory.

> --binary-version=5.6.48
> 
> If the above assumption is correct, this option is not needed, as it says
> "the directory /gome/mikiya/opt/ndb/7.4.28" contains version 5.6.48

I see. Is the actual version number retrieved from mysqld?

> --my-cnf-options='server_id=1'
> 
> there is an option --server-id that does this

I see.

> --my-cnf-options='!include /home/mikiya/sandboxes/cluster_7_4_28/my_common.cnf'
> 
> Depending on what is in this include file, anything can go wrong.
> dbdeplpoyer has an option --my-cnf=file_name to use an external my.cnf more
> safely

I see. I'll modify my program to do so.

> --flavor=ndb
> 
> Not needed for deployment

I see.

My suggestion is to let dbdeployer to employ mysql_install_db for MySQL NDB Cluster versions <= 7.4. It will work just fine.

Kind regards,
Mikiya

**Comment by datacharmer** [comment]
Fixed in version 1.53.0
